keps/sig-scheduling/NNNN-pod-to-node-assignment/README.md

```diff
@@ -0,0 +1,808 @@
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Customers trying to deploy batch workloads on Kubernetes have filed multiple
+requests (See [1](https://github.com/kubernetes/kubernetes/issues/68827),
+[2](https://github.com/kubernetes/kubernetes/issues/105287) for details) for 1:1
+pod-to-node assignment.
+
+Currently there are two approaches to achieve this:
+
+-   set resource requests to something high enough to prevent scheduling any
+    other pods but daemonset pods. This is tricky to do and error prone,
+    especially for smaller nodes.
+-   Use pod anti affinity. Users will need to define the same set of labels on
+    all pods that needs to be repelled, whether or not they actually request 1:1
+    pod to node assignment, it is also expensive to calculate especially for
+    cluster autoscaler.
+
+A simple and native way to provide 1:1 pod-to-node assignment is needed to be
+supported by the scheduler.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+-   Allow users to force 1:1 pod-to-node assignment for a given pod
+
+## Proposal
+
+The idea is to create a scheduler filter plugin such that given a pod and a
+node, a node is filtered out if any of the following is true (assume the API is
+the annotation k8s.io/pod-per-node=true, more on that below):
+
+-   The incoming pod is not a daemonset pod (by checking OwnerReference) and
+    there is a pod with k8s.io/pod-per-node=true already assigned to the node,
+-   If the incoming pod has the k8s.io/pod-per-node=true annotation and the node
+    hosts at least one non daemonset pod. The above logic would also be added to
+    the set of predicates that kubelet tests on pod admission. This will prevent
+    users from working around the scheduler to avoid 1:1 enforcement.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+A user wants a Pod to be scheduled alone on a node to avoid them from contending
```

Comment:
I would replace this story with something a bit more concrete: `as a user, I want to deploy a Spark or MPI job such that only a single worker is running on each pod excluding daemonsets.`

Reply:
Would it make sense to keep both (a story that gives an idea of why the feature could be used, and the other one would be more concrete/detailed)?

Maybe we can merge them into a single story?